sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirSearchParameter.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.healthcare;
+
+import java.util.Map;
+import java.util.Objects;
+import org.apache.beam.sdk.coders.DefaultCoder;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * FhirSearchParameter represents the query parameters for a FHIR search request, used as a
+ * parameter for {@link FhirIO.Search}.
+ */
+@DefaultCoder(FhirSearchParameterCoder.class)
+public class FhirSearchParameter<T> {
```

Comment:
will the queries within the `Map` usually be of the same type for a single object and for a whole PCollection? I had understood that for any individual pcollection, these could be very different (Enums, Strings, Numbers?)

Reply:
Is it possible to create a PCollection with a variation of types like that with a coder?

In my use case and I think most expected use cases, you would use only one (I only use List<String>).

Reply:
yeah, not really. I think this makes sense (I like it much better than just setting Object, and leaving everything else to the user defining a coder)